### PR TITLE
Add `Icon` prop to component

### LINKED ISSUE #104: Add custom `Icon` component as prop
**Is your feature request related to a problem? Please describe.**
At work, we would like to be able to add a custom icon component to be rendered by the picker select.

**Describe the solution you'd like**
The solution is simple, just add an extra `prop`, called `Icon` to be rendered instead of the current `Icon` used.

**Describe alternatives you've considered**
I couldn't think of any other way of solving this in this context, apart from the current way, which is to style the view that is used as an Icon.

**Additional context**
Here is a screenshot of how we would like the component to look.
screen shot 2018-10-23 at 17 58 28

### PR DESCRIPTION
**Pull request by luizParreira**
- Adding the possibility of the user passing a custom `Icon` prop to be
rendered instead of the default one
- Added README.md instructions
- Add tests for the new behaviour

Closes #104

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -416,6 +423,7 @@ export default class RNPickerSelect extends PureComponent {
                 >
                     {this.renderPickerItems()}
                 </Picker>
+                {Icon && this.renderIcon()}
```
Comment (lfkwtz) [inline src/index.js]: not a fan of this syntax personally -- would prefer:

`{Icon ? this.renderIcon() : null}`

**Comment by luizParreira** [comment]
@lfkwtz the commit ce056cc has made the changes requested  😃

**Comment by balthazar** [comment]
Any plan to get this merged? Since @luizParreira already did the testing, adding one picker using the new prop in the `example` file should be enough no?

**Comment by lfkwtz** [comment]
I tested it myself locally and it wasn’t working as I imagine it should - which is why I’m asking for working examples

**Comment by balthazar** [comment]
I see, makes sense 👍

**Comment by lfkwtz** [comment]
@luizParreira

**Comment by lfkwtz** [comment]
Going to close this out as I haven't heard from you in a couple months. We can re-open later.

**Pull request closed without merging**